ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/jsonrpc/JsonRpcHttpServiceTlsTest.java

```diff
@@ -83,25 +86,19 @@
   private static final BigInteger CHAIN_ID = BigInteger.valueOf(123);
   private static final Collection<RpcApi> JSON_RPC_APIS = List.of(ETH, NET, WEB3);
   private static final NatService natService = new NatService(Optional.empty());
-  private static final String ROOT_RESOURCE = "JsonRpcHttpService/";
-  private static final Path KEYSTORE_PATH =
-      Paths.get(getResource(ROOT_RESOURCE + "rpc_keystore.pfx").getPath());
-  private static final Path KEYSTORE_PASSWORD_FILE =
-      Paths.get(getResource(ROOT_RESOURCE + "rpc_keystore.password").getPath());
-  private static final Path KNOWN_CLIENTS_FILE =
-      Paths.get(getResource(ROOT_RESOURCE + "rpc_known_clients.txt").getPath());
-  private static final Path CLIENT_CERT_KEYSTORE_PATH =
-      Paths.get(getResource(ROOT_RESOURCE + "rpc_client_keystore.pfx").getPath());
-  private static final Path X_CLIENT_CERT_KEYSTORE_PATH =
-      Paths.get(getResource(ROOT_RESOURCE + "rpc_client_2.pfx").getPath());
-
   private JsonRpcHttpService service;
   private String baseUrl;
   private Map<String, JsonRpcMethod> rpcMethods;
   private final JsonRpcTestHelper testHelper = new JsonRpcTestHelper();
+  private SelfSignedPfxStore selfSignedPfxStore;
+  private SelfSignedPfxStore selfSignedPfxStoreForHttpClient;
+  private SelfSignedPfxStore selfSignedPfxStoreForHttpClientNotTrustedByServer;
 
   @Before
   public void initServer() throws Exception {
+    selfSignedPfxStore = SelfSignedPfxStore.create();
+    selfSignedPfxStoreForHttpClient = SelfSignedPfxStore.create();
+    selfSignedPfxStoreForHttpClientNotTrustedByServer = SelfSignedPfxStore.create();
```

Comment:
This seems odd, the same method of creation gives a `PfxStore`, `PfxStoreForHttpClient` and `PfxStoreForHttpClientNotTrustedByServer`. 

If there are three different stores, then perhaps separate helper methods to create each of them and their corresponding state (e.g. that makes one trusted and another not trusted), might be an idea?

Reply:
The utility class generate both keystore and truststore containing self-sligned certificate. The keystore is used by server/json-rpc and truststore is used by okhttp/client vice-versa is also applicable for testing client-auth. The keystore used by okhttp client, its truststore is used by server. Hence there is a need of having three different instances of self-signed certs. I can wrap them in a method for the sake of reaability, but all it will do is return instances which will again store in three different variables.